beacon-chain/rpc/validator/proposer.go

```diff
@@ -176,6 +274,42 @@ func (vs *Server) ProposeBlock(ctx context.Context, rBlk *ethpb.SignedBeaconBloc
 	}, nil
 }
 
+// ProposeBlockV2 is called by a proposer during its assigned slot to create a block in an attempt
+// to get it processed by the beacon node as the canonical head.
+func (vs *Server) ProposeBlockV2(ctx context.Context, rBlk *ethpb.SignedBeaconBlockAltair) (*ethpb.ProposeResponse, error) {
+	blk := interfaces.WrappedAltairSignedBeaconBlock(rBlk)
+	root, err := blk.Block().HashTreeRoot()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not tree hash block: %v", err)
+	}
+
+	// Do not block proposal critical path with debug logging or block feed updates.
```

Comment:
The block feed, broadcast and pass to receive block can be shared between `ProposeBlock`